_posts/_2021-08-31-building-opensearch-1.1-distributions.markdown

```diff
@@ -0,0 +1,44 @@
+
+This worked well until the [OpenSearch fork](https://aws.amazon.com/blogs/opensource/introducing-opensearch/). It was decided that OpenSearch would be a product with two distributions at launch: a full distribution that includes a dozen plugins, and a minimal distribution without any plugins. This made sense because most customers either wanted a well tested distribution with everything, or are technically savvy enough to assemble their favorite flavor of all the things. 
+
+The full distribution of OpenSearch was similar to ODFE, but with one major difference - the OpenSearch service would be developed in parallel with the plugins. The complete OpenSearch distribution became effectively a monolith spanning multiple GitHub repositories. You just can't both develop all the moving parts simultaneously, and have stable dependencies of [OpenSearch](https://github.com/opensearch-project/OpenSearch), [common-utils](https://github.com/opensearch-project/common-utils), [job-scheduler](https://github.com/opensearch-project/job-scheduler), to name a few, at the same time.
+
+To produce a reliable monolith that can be trusted we decided to build the entire distribution from source. We developed a [manifest-based workflow](https://github.com/opensearch-project/opensearch-build/tree/main/bundle-workflow) to accomplish this, which consists of the following steps.
+
+1. Take an input manifest ([example](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/opensearch-1.1.0.yml)), build all components from source, and generate a build manifest. Today we build x64 and arm64 flavors.
+2. Take a build manifest ([example](https://github.com/opensearch-project/opensearch-build/blob/main/bundle-workflow/tests/tests_manifests/data/opensearch-build-1.1.0.yml)) and assemble a distribution.
+3. Sign the artifacts.
+4. Run integration, backwards-compatibility, performance, and upgrade tests against the assembled distribution.
+5. Publish maven artifacts to our snapshot maven repository and the distributions to S3.
+
+Here's all the inputs and outputs for OpenSearch 1.1.
+
+* Maven releases are staged in the [AWS OSS Maven Repository](https://aws.oss.sonatype.org/content/repositories/releases/org/opensearch/) before being promoted to Maven Central. 
+* TODO
+
+You can reproduce the build and test process by checking out [opensearch-build](https://github.com/opensearch-project/opensearch-build/) and running it yourself. You will need additional tools, including `maven` and `cmake`, to compile some of the components.
+
+```
+./bundle-workflow/build.sh manifests/opensearch-1.1.0.yml
+./bundle-workflow/assemble.sh manifests/opensearch-1.1.0.yml
+
+TODO: test
+```
+
+We're in the process of reusing this system for OpenSearch Dashboards, see [opensearch-build#158](https://github.com/opensearch-project/opensearch-build/issues/158).
```

Comment:
Love the callout on next steps, almost feels like we should be referencing a roadmap that is externally visible.